### ISSUE TITLE
Dash to dock broken, unable to rearrange icons

### ISSUE THREAD

**Issue by coreybruce** [micheleg/dash-to-dock#2010]
Hey there I wanted to report a issue with dash to dock, I am unable to rearrange the icons using dash to dock and the only way  I was able to make it work was to set it to panel mode and also resizing icons doesn't not work unless it's set to panel mode. and have no idea why it's happening

I tried 
```
dconf reset -f /org/gnome/shell/extensions/dash-to-dock/
```

But it is still broken.

[Screencast from 2023-05-17 02-58-17.webm](https://github.com/micheleg/dash-to-dock/assets/8393780/8964f0fd-6324-4719-b746-73dc1d29c645)

I am using Manjaro Gnome using the latest version 81 and tested both the extension package from Manjaros repo and from Gnome extensions and have the same issue.

**Comment by doppelhelix** [comment]
Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock"
That does the trick for me.

**Comment by Fascomino** [comment]
When I moved the Show Applications icon to the end everything worked as intended.  This error was introduced with the last change to put the button at the edge when in panel mode.

**Comment by ionutbortis** [comment]
Same here. No rearranging and the new opened apps that were not previously pinned to dash as favorites are not displayed properly.
[dash to dock 81 issues.webm](https://github.com/micheleg/dash-to-dock/assets/5013697/fe3e69e3-2412-4d44-ad68-4ede0c25f9c0)

**Comment by coreybruce** [comment]
> Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock" That does the trick for me.

Hmm yeah that also does it but what is causing this issue I wonder?

**Comment by coreybruce** [comment]
This is only happening on one machine, my laptop for example isn't sufferering from this same issue.

**Comment by ionutbortis** [comment]
> > Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock" That does the trick for me.
> 
> Hmm yeah that also does it but what is causing this issue I wonder?

There are some recent changes that affect the dock behavior. I downgraded to version 80 and changed the version from the local `metadata.json` file to 81 in order to avoid future update to the problematic version.

**Comment by coreybruce** [comment]
If anyone is using a Arch based system like me which uses Manjaro I have the backup package of the 80-1 package from Manjaro, I know this isn't recommended or trust worthy but I have the sig file to verify it

**Comment by ionutbortis** [comment]
@FlyingPaperPlane Yeah, I hope these issues will get fixed in the 82 version release :pray: 

If not, we can redo the above downgrade process and change version from 81 to 83 :sweat_smile:

**Comment by vikingredwolf** [comment]
Update: I noticed you can't rearrange icons when the apps icon is on the left, but works perfectly when it's on the right side. Isn't that odd?

**Comment by fraz0815** [comment]
> Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock" That does the trick for me.

Thanks, confirming the workaround on manjaro gnome 44.1 testing.

**Comment by jcsekinger** [comment]
> Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock" That does the trick for me.

and re-enable ;)

**Comment by coreybruce** [comment]
> > Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock" That does the trick for me.
> 
> and re-enable ;)

Sadly that doesn't work, if you re-enable it than it will go back to the same issue.

**Comment by jcsekinger** [comment]
> > > Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock" That does the trick for me.
> > 
> > 
> > and re-enable ;)
> 
> Sadly that doesn't work, if you re-enable it than it will go back to the same issue.

yes, it does not really solve the pb, it is juste a workaround: disable > move the icon > re-enable it  — _and the icon is now at the correct place, that’s all_ you have to do it every time you have to move an icon.

**Comment by coreybruce** [comment]
> > > > Same Problem here. the workaround is to disable "Move the applications button at the beginning of the dock" That does the trick for me.
> > > 
> > > 
> > > and re-enable ;)
> > 
> > 
> > Sadly that doesn't work, if you re-enable it than it will go back to the same issue.
> 
> yes, it does not really solve the pb, it is juste a workaround: disable > move the icon > re-enable it — _and the icon is now at the correct place, that’s all_ you have to do it every time you have to move an icon.

Yeah I agree

**Comment by coreybruce** [comment]
Any progress on fixing this issue?

**Comment by 3v1n0** [comment]
https://github.com/micheleg/dash-to-dock/pull/2021 should fix it, please test!

**Comment by jcsekinger** [comment]
> 

how can I test that solution ? How do I  « Use a different container to hold floating ShowAppsIcon»? This is not significant for me. Thank you !

**Comment by ionutbortis** [comment]
> #2021 should fix it, please test!

Hi Marco,

Thanks for looking into this!

I tested the [rearrange-fix-icons](https://github.com/3v1n0/dash-to-dock/tree/rearrange-fix-icons) branch, we can now rearrange the icons, that's good. 

But on my side, there's still a small issue:
When you open an app that is not already added to the favorites, it will be added at the end, with a separator. All newly opened apps after that one are opened to its left, not to its right, as it was functioning before.

Please see the attached screencast.
[Screencast from 2023-05-31 15-23-40.webm](https://github.com/micheleg/dash-to-dock/assets/5013697/fca8b5b8-7034-4c5d-9b0b-9fe26880d9f9)

**Comment by 3v1n0** [comment]
> All newly opened apps after that one are opened to its left, not to its right, as it was functioning before.

I think this bug was also present before v80 and so it's another issue, but thanks for letting me know

**Comment by Amzd** [comment]
Just tested after #2021 merge and I think this is related:

Last pinned item moves to the wrong side of the separator when a different item is unpinned. 

[Screencast from 2023-06-02 04-07-10.webm](https://github.com/micheleg/dash-to-dock/assets/17001151/95205164-fa09-4292-87d8-0972847ba60f)

**Comment by 3v1n0** [comment]
Thanks @Amzd, please test https://github.com/micheleg/dash-to-dock/pull/2023

**Comment by Amzd** [comment]
unpin edge case is fixed :+1: seems to all work fine in #2023 with applications button on the left @3v1n0

**Comment by ionutbortis** [comment]
> > All newly opened apps after that one are opened to its left, not to its right, as it was functioning before.
> 
> I think this bug was also present before v80 and so it's another issue, but thanks for letting me know

You're right, that bug seems to be introduced in v80. Just tested v79 and the issue is not present there.

Also, #2023 fixes that issue, the newly opened apps that are not pinned to the dash are properly getting their order in the dash. Thanks!

**Comment by eboye** [comment]
I stumbled upon this issue as I'm having the same behavior on Gnome 45.1 and D2D v89

[Screencast from 2023-11-18 22-11-35.webm](https://github.com/micheleg/dash-to-dock/assets/624357/b2fa5a90-1a6f-4db0-8772-3bf225cb3b39)

**Comment by vanvugt** [comment]
This bug is closed, but you can continue the conversation in https://github.com/micheleg/dash-to-dock/issues/2122

**Comment by brunos3d** [comment]
After researching the issue, I came across multiple discussions regarding the inability to display pinned apps in the AppGrid without affecting the layout of the Dash-to-Dock extension. This led me to fork the original [favourites-in-appgrid](https://gitlab.gnome.org/harshadgavali/favourites-in-appgrid/) extension (in which causes the re-arrangement issue) and implement a solution that would restore the visibility of pinned apps in both the GNOME Dock and Application Grid, without disrupting any existing Dock arrangements.

https://github.com/brunos3d/pinned-apps-in-appgrid

This repository aims to provide a simple yet effective fix to this issue, preserving the GNOME Dock layout while ensuring pinned apps remain visible in the grid.

**Comment by brunos3d** [comment]
Our extension is finally available on gnome extensions
https://extensions.gnome.org/extension/7660/keep-pinned-apps-in-appgrid/

**Comment by xahitdev** [comment]
I was having a problem where I put my PC into sleep and everything breaks afterwards. 
In the "Extensions" there is  that setting called "ubuntu dock" if you click settings and customize it, it eventually becomes better than dash to dock extension  and doesn't break the system.
The Ubuntu Dock comes pre-enabled within Ubuntu, no need for Dash To Panel or Dash To Dock for myself at least.

**Comment by vanvugt** [comment]
> I was having a problem where I put my PC into sleep and everything breaks afterwards.

Sounds like #2179. But that should be [fixed in Ubuntu](https://bugs.launchpad.net/bugs/2069559) at least.

Also this bug is closed. Please open new issues for any problems you find.
